### ISSUE TITLE
[bug] vm extensions module causing huge delay in terraform plan

### ISSUE THREAD

**Issue by brk3** [Azure/caf-terraform-landingzones#343]
**Describe the bug**
It seems the following two files are increasing the time for a terraform plan by a factor of 6:

* https://github.com/Azure/caf-terraform-landingzones/blob/main/caf_solution/vm_extensions.tf
* https://github.com/Azure/caf-terraform-landingzones/blob/main/caf_solution/vmss_extensions.tf

An empty landingzone plan currently takes ~ 12 minutes on my machine (late 2013 macbook pro, 2.4 GHz Dual-Core Intel Core i5), while with the above files removed it takes ~ 2 minutes.

**To Reproduce**
* Deploy a launchpad
* In any level, create a "passthrough" landingzone, i.e. a no config other than the following:
```
landingzone = {
  backend_type        = "azurerm"
  global_settings_key = "launchpad"
  level               = "level1"
  key                 = "passthrough"
  tfstates = {
    launchpad = {
      level   = "lower"
      tfstate = "caf_launchpad.tfstate"
    }
  }
}
```
* Run a plan and time it:
```
time /tf/rover/rover.sh \
  -lz /tf/caf/landingzones/caf-terraform-landingzones/caf_solution/ \
  -var-folder /tf/caf/configuration/level1/passthrough/ \
  -level level1 \
  -env ${ENV} \
  -tfstate passthrough.tfstate \
  -a plan
```
* Now in caf_solution, remove the two files referenced above, and rerun the test.

**Expected behavior**
I expect landingzone deploys to complete in a reasonable amount of time (< 5 mins) when there are few or no resources to be deployed.

**Additional context**
I'd like to get some more context on what `vm_extensions.tf` /  `vmss_extensions.tf` are used for, and how we can integrate them in a better way. Both are depending on `module.solution` which is not recommended and likely the source of this issue.

**Comment by arnaudlh** [comment]
Planned to be fixed in module 5.6.0 as you can see there https://github.com/aztfmod/terraform-azurerm-caf/issues/968

**Comment by brk3** [comment]
Thanks @arnaudlh. Does the above work also address the issue in `caf_solution`? Performance optimizations in terraform-azurerm-caf are very welcome but this particular issue is to do with `caf_solution`.

**Comment by arnaudlh** [comment]
Yes @brk3 for now it is only in caf_example for incubation but it will be in caf_solution as well. 
Now onwards, we are going to create integration branches typically ```integration-2204``` ish. So it will integrate that in caf_solution in the next few days.

**Comment by SebTardif** [comment]
99% of my time is now waiting. Is this fixed in main branch?

**Comment by shuft** [comment]
Agreed, we're waiting a loooong time for plans these days - not good :(

Any workaround whilst we (continue to :)) wait?

**Comment by LaurentLesle** [comment]
Closed as completed with 5.7.x
